File: factory/CarteBancaire.java
package factory;

public class CarteBancaire extends LocalPaymentMethod {

    public CarteBancaire(String cardHolder, String cardNumber, String cvv, String expiryDate) {
        super(cardHolder, cardNumber, cvv, expiryDate);
    }

    @Override
    public void authorizePayment() {
        System.out.println("Authorizing Carte Bancaire card....");
    }

    @Override
    public void startMoneyTransfer() {
        System.out.println("Start money transferring for a Carte Bancaire card....");
    }

    @Override
    public void calculatePaymentFees(double amount) {
        System.out.println("Calculating Carte Bancaire card payment fees....");
    }
}

File: typescript/factory/carte-bancaire.ts
import { LocalPaymentMethod } from "./local-payment-method";

export class CarteBancaire extends LocalPaymentMethod {
    constructor(
      cardHolder: string,
      cardNumber: string,
      cvv: string,
      expiryDate: string
    ) {
      super(cardHolder, cardNumber, cvv, expiryDate);
    }
  
    authorizePayment(): void {
      console.log("Authorizing Carte Bancaire card....");
    }
  
    startMoneyTransfer(): void {
      console.log("Start money transferring for a Carte Bancaire card....");
    }
  
    calculatePaymentFees(amount: number): void {
      console.log("Calculating Carte Bancaire card payment fees....");
    }
  }